### DISCUSSION TITLE
How can i add existing object to entity?

### DISCUSSION THREAD

**Discussion by sametydev** [skypjack/entt#1195] [Q&A]
Hello,
I have exist object Object but in usage of emplace, its creating new object with generic type.
Any ways to can i add existing object to entity?

**Comment by skypjack** [comment]
You can use `std::move`. A storage manages its memory, you have to transfer the data there somehow.

**Reply by sametydev** [reply]
Actually now, I have an interface called Script, DLLs generate objects from this Script (Like MyScript, or PlayerScript but my program read this as Script),
I can access these objects as pointers, so I guess using std::move may not move the data completely,
is there any way to add the existing class object directly?

**Reply by skypjack** [reply]
No, I'm sorry. Again, a storage manages its memory and it doesn't allocate a new chunk of memory for every element. So, there isn't a way to pass in your data without moving or copying them into the storage. You can make your script generate the objects into the registry directly though, I don't see the problem. It's pretty common.
